Text Generation WebUI/src/order-helpers.ts

```diff
@@ -0,0 +1,197 @@
+
+  return finishStatuses.includes(orderInfo.status);
+};
+
+const getOrderEncryptedResult = async (
+  orderId: string,
+  logger: Logger,
+): Promise<EncryptedOrderResult> => {
+  logger.debug('Getting order encrypted result');
+  const connector = BlockchainConnector.getInstance();
+
+  try {
+    await connector.initialize({
+      blockchainUrl: config.blockchainUrl,
+      contractAddress: config.blockchainContractAddress,
+    });
+
+    const retryIntervalInSeconds = 10;
+    await helpers.tryWithInterval({
+      handler() {
+        logger.debug('Checking if order is finished');
+        return isOrderFinished(orderId);
+      },
+      checkResult: (isResultOk) => ({ isResultOk }),
+      retryInterval: retryIntervalInSeconds * 1000,
+      retryMax: (60 * 60 * 3) / retryIntervalInSeconds,
+    });
+
+    const order = new Order(orderId);
+    const orderInfo = await order.getOrderInfo();
+
+    if (orderInfo.status === OrderStatus.Canceled) {
+      throw new Error(`Tunnel Provisioner Order is canceled (orderId=${orderId})`);
+    }
+    if (orderInfo.status === OrderStatus.Error) {
+      throw new Error(`Tunnel Provisioner Order failed (orderId=${orderId})`);
+    }
+
+    const { encryptedResult } = await order.getOrderResult();
+
+    logger.debug('Order encrypted result received');
+
+    return JSON.parse(encryptedResult) as EncryptedOrderResult;
+  } finally {
+    connector.shutdown();
+  }
+};
+
+const decryptOrderResult = async (params: {
+  encryptedResult: EncryptedOrderResult;
+  orderKey: EncryptionKey;
+}): Promise<{ resource: StorageProviderResource; encryption: Encryption }> => {
+  const { encryptedResult, orderKey: encryptionKey } = params;
+
+  const publicKeyEncryption = Crypto.getPublicKey(encryptionKey);
+  const derivedPrivateKey = await RIGenerator.getDerivedPrivateKey(publicKeyEncryption);
+
+  const decryptedResourceStr = await tryDecryptWithKeys(encryptedResult.resource, [
+    encryptionKey.key,
+    derivedPrivateKey.key,
+  ]);
+  if (!decryptedResourceStr) {
+    throw new Error('Could not decrypt resource');
+  }
+
+  const decryptedEncryptionStr = await tryDecryptWithKeys(encryptedResult.encryption, [
+    encryptionKey.key,
+    derivedPrivateKey.key,
+  ]);
+  if (!decryptedEncryptionStr) {
+    throw new Error('Could not decrypt encryption');
+  }
+
+  return {
+    resource: JSON.parse(decryptedResourceStr) as StorageProviderResource,
+    encryption: JSON.parse(decryptedEncryptionStr) as Encryption,
+  };
+};
+
+const tryDecryptWithKeys = async (
+  encryption: Encryption,
+  decryptionKeys: string[],
+): Promise<string | undefined> => {
+  for (const decryptionKey of decryptionKeys) {
+    try {
+      return await Crypto.decrypt({
+        ...encryption,
+        key: decryptionKey,
+      });
+    } catch (e) {
+      continue;
+    }
+  }
+};
+
+export const getOrderResult = async (params: DownloadOrderResultParams): Promise<Readable> => {
```

Comment:
move to sdk. We can make use of it in spctl download result command to automatically unpack

Reply:
It will be better to implement something like:
```ts
import { OrdersService } from '@super-protocol/spctl-core';
// ...
const orderId = '3701';
const service = new OrdersService();
const result = await service.getOrderResult(orderId);
```

I am depressed by the idea of ​​moving these functions to the sdk. I will be forced to add new dependencies such as `tar-stream`. Ideally, the sdk should export only methods for working with contracts. More complex methods should be in other packages.